### PR TITLE
feat(ui): integrate thread api in Answer Engine

### PR DESCRIPTION
**Pull request by liangfung**
### Screen record
https://jam.dev/c/b29b8a8c-eda9-46c8-9c40-8786c94f3b39

### CODE REVIEW

**Comment by wsxiaoys** [comment]
Please refer perplexity for url format: 
1. new search: `https://www.perplexity.ai/search/new?q=what+does+chat_completions+do?`
2. thread: `https://www.perplexity.ai/search/what-does-chat-completions-do-lTi4KAufSHCj.eQKRinPHQ`

File: ee/tabby-ui/lib/tabby/gql.ts [hunk 1]
```diff
@@ -342,7 +348,27 @@ const client = new Client({
...
     }),
-    fetchExchange
+    fetchExchange,
+    subscriptionExchange({
+      forwardSubscription(request, operation) {
+        const authorization =
+          // @ts-ignore
+          operation.context.fetchOptions?.headers?.Authorization ?? ''
+        const wsClient = createWSClient({
+          url: 'http://localhost:8080/subscriptions',
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/tabby/gql.ts:358]:
```suggestion
          url: '/subscriptions',
```

File: ee/tabby-ui/app/search/components/search.tsx [hunk 2]
```diff
@@ -120,79 +136,156 @@ export const SearchContext = createContext<SearchContextValue>(
...
+        .map(o => o.node)
+        .slice()
+      setMessages(messages)
+      setIsReady(true)
+    }
+  }, [threadMessages])
+
+  const onThreadCreated = (threadId: string) => {
+    updateUrlComponents({
+      pathname: `/search/${threadId}`
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/search/components/search.tsx:255]:
```suggestion
      pathname: `/search/${threadId}`
```

pathname shall be something like `/search/what-does-chat-completion-do-${threadId}`

**Comment by wsxiaoys** [comment]
image

Message completion is not handled properly - `Stop Generation` button still presents after message is finished.

File: ee/tabby-ui/app/search/components/search.tsx (hunk 2 above)
Comment (liangfung) [reply ee/tabby-ui/app/search/components/search.tsx]: done

**Pull request merged**